src/containers/tud/components/QuestionnaireForm.js

```diff
@@ -58,6 +58,50 @@ const ButtonRow = styled.div`
   margin-top: 30px;
 `;
 
+/*
+ * This code fixes an issue where the survey enters an error state after a user fills out the survey
+ * and later changes the day end time value. If the day end time is changed such that certain sections
+ * of the formRef state contain invalid data (the start/end values are out of range), then this will cause
+ * validateAndSubmit to fail, hence the need to remove such sections. This deletion need only take place
+ * just before we enter the nightActivity portion of the survey. At this point, for all x s.t x > page
+ * only x == page + 1 can be expected to contain any data
+  (and it should be nightActivity data otherwise it also needs to be deleted)
+ */
+
+const removeExtraData = (formRef :Object, pagedData, page :number) => {
+  const psk = getPageSectionKey(page, 0);
+  const dayEndTime :?DateTime = selectTimeByPageAndKey(DAY_SPAN_PAGE, DAY_END_TIME, pagedData);
+  const formData :Object = formRef?.current?.state?.formData || {};
+
+  const currEndTime = formData[psk]?.[ACTIVITY_END_TIME];
+
+  if (dayEndTime && currEndTime) {
+    const currEndDateTime :DateTime = DateTime.fromSQL(currEndTime);
+    if (!currEndDateTime.equals(dayEndTime)) {
+      return;
+    }
```

Comment:
Wait, don't you want to proceed if they are not equal?

> and later changes the day end time value

So if they change the value, whatever is in `formData` would be different, no? In which case, since the values are different, wouldn't you want to proceed?

Reply:
we only need to do this "forced mutation before submit" workaround when the user indicates that they are done filling the daytime part of the survey (that is when dayEndTime === endTime). Therefore, if they are not equal we don't need to mutate anything; we just execute Paged's validateAndSumit run as usual.

Reply:
Got it. Thanks!